webpack/assets/javascripts/react_app/routes/ForemanSwitcher.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Switch } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { selectComponentByWeight } from '../components/common/Slot/SlotSelectors';
+
+export const ForemanSwitcher = ({ children }) => {
+  const externalRoutes = selectComponentByWeight('routes') || [];
+
+  return <Switch>{externalRoutes.concat(children)}</Switch>;
+};
+
+// map state to props
+const mapStateToProps = state => ({
+  routes: state.extendable.routes || [],
```

Comment:
Am i missing something? I don't see any `routes` prop in the `ForemanSwitcher` component?

Reply:
right... forgot the condition :sweat_smile: